### ISSUE TITLE
"Plate to plate" operation cuts are not recognized by Grasshopper components and "Deconstruct Cut" flattens data

### ISSUE THREAD

**Issue by tbyrk** [idea-statica/ideastatica-public#518]
I'm using 24.0.1.1233 and 0.1.6.

I have three connections here that are fed into the Grasshopper components at the same time.

On the left is N3 with a "cut by work plane" and "End plate" operation.
Middle is N4 with a "Plate to plate" operation.
On the right is N5 with a "Plate to plate" operation.

<img width="1274" alt="PP - 1" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/16ed7c9e-a5c6-45b9-9345-0c1b0bf72845">

This one is the "Plate to plate" operation.

<img width="1136" alt="PP - 2" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/cceea9d6-4ca3-4016-9439-52f0604d62cb">

In Grasshopper we see that only the "End plate" operation cut exists. "Plate to plate" operation cuts are missing. "Plate to plate" operation should create two cuts. One for each member.

<img width="1274" alt="PP - 3" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/3ddbc227-8357-4f51-b79a-6c9a3088dc20">

These connections are going through the following pathway:

1. Reference Connection
2. Extract IOM Container
3. Modify Open Model
4. Deconstruct Open Model

<img width="1157" alt="PP - 4" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/b2c55e0c-bfa4-4527-9973-c82fe7f06e80">

Connections/SAF/script are attached.

Second issue is that there is some error with the "Deconstruct Cut" component although it does give me the "CM3 is cut by plate EP2" which is actually correct for connection N3. It also appears to force flatten data which means plate cut data trees are not properly generated which means it cannot be used for bulk connection geometry generation for Tekla.

[PP - Connections.zip](https://github.com/idea-statica/ideastatica-public/files/15370086/PP.-.Connections.zip)

**Comment by nathanlukeIDEA** [comment]
Hi @tbyrk,

You are correct. I am transferring this to the development team. I will get back to you. 

Cheers
Nathan

**Comment by nathanlukeIDEA** [comment]
@tbyrk, Can you please now try with the latest release? (24.0.2) We have added some additional cut outputs. It is getting flattened incidentally by a grasshopper. The component runs until it hits the null reference error.

**Comment by tbyrk** [comment]
Before I get to the stiffening members I stumbled upon this. There is no "Location" output now. How to get the point coordinates from the Goo object "Ref Point"?

![image](https://github.com/idea-statica/ideastatica-public/assets/162557189/d0078bf1-7deb-4512-8f8f-fe25bd57c0ef)

**Comment by tbyrk** [comment]
I'm still having issues with the "Deconstruct Connected Member" cut planes data tree structure. I have three connections in the input. All other components understand that it means three branches of data but cut planes output gives each member a separate branch (two members per connection = six branches in this case).

For example for anything else the script understands that all connection 1 elements should be moved by (6000,0,7000), connection 2 elements by (0,0,5000) and connection 3 elements by (12000,0,5000). But since the cut plane output has a separate branch for each member then the connection 2 ID=1 element (third in the list) gets moved by connection 3 coordinates instead of connection 2 coordinates.

<img width="943" alt="image" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/d685e154-431c-48e0-9067-384b07dd3676">


<img width="975" alt="image" src="https://github.com/idea-statica/ideastatica-public/assets/162557189/1ce56452-dfbd-4aab-bbe6-e223414aaf95">

**Comment by nathanlukeIDEA** [comment]
Hi @tbyrk, I have updated a new version that allows casting the point ref to a Rhino point. Regarding the cuts. This is more of an issue on the IOM data structure rather than a grasshopper issue. We will hopefully improve the way we are storing plane cuts in IOM - I don't want to go away from IOM for the time being. The problem is that a connected member can have multiple cuts (list) which causes appending of branches for each member.   See attached and a script below. Regarding the planes for the member cuts. Plate View outputs the front and back planes already I added this to the script aswell. 
([PP - Script_nl.zip](https://github.com/user-attachments/files/16022936/PP.-.Script_nl.zip))
![image](https://github.com/idea-statica/ideastatica-public/assets/93578548/ad2f1139-320f-4d10-ae2b-1d10ca4ebfe3)
